app/src/main/java/com/karumi/data/repository/SuperHeroRepository.kt

```diff
@@ -1,140 +1,18 @@
 package com.karumi.data.repository
 
-import android.util.Log
+import com.karumi.domain.model.DomainError
 import com.karumi.domain.model.SuperHero
+import org.funktionale.either.Either
 
-class SuperHeroRepository {
-    private val superHeroes: List<SuperHero>
+class SuperHeroRepository(private val datasources: List<SuperHeroDataSource>) {
 
-    companion object {
-        private val BIT_TIME = 1500L
-        private val TAG = "SuperHeroRepository"
+    fun getAllSuperHeroes(): Either<DomainError, List<SuperHero>> {
+        val values = datasources.first { it.isUpdated() }.getAll()
+        values.component2()?.let { list -> datasources.forEach { it.populate(list) } }
```

Comment:
This will repopulate the cache even if it gets the information right from the cache, is this what we want?

Reply:
yes, I was thinking about this one, I take this solution only to simplify the use case.